### DISCUSSION TITLE
Can we use publicPath a relative path and not baseUrl?

### DISCUSSION THREAD

**Discussion by iamprobirdey** [module-federation/module-federation-examples#1636] [Q&A]
Hi,

I am using module federation with vue 3.

**Child:**
If I add `publicPath: http:localhost:8082` and run the server

I can get remote file which I exposed to my parent app.

**Problem 1:**

If I add `publicPath: http:localhost:8082` then on my Child server I noticed it runs on

`http//locahost:8082/http://localhost:8082/` path.

**Problem 2:**

if I don't add the complete baseUrl on the child app
I don't receive remote files.

**question**
Can I use `publicPath: '/'` and aslo get the remote files.

Thanks.

**Comment by iamprobirdey** [comment]
@ScriptedAlchemy Any comment on this would be appreciated. :)

**Comment by XiaofengXie16** [comment]
Yes, you can change the publicPath to "/", you can also try `publicPath: "auto"'.

In addition to that , you have three approaches to change the path dynamically at runtime.
1. use dynamic: host https://github.com/module-federation/module-federation-examples/tree/master/dynamic-system-host
2. use external-remotes-plugin :https://github.com/module-federation/module-federation-examples/issues/566
3. use promise new promise syntax: https://webpack.js.org/concepts/module-federation/#promise-based-dynamic-remotes

similar issue: https://github.com/module-federation/module-federation-examples/discussions/577#discussioncomment-2203510

**Reply by iamprobirdey** [reply]
Hey,
Thanks for writing down.

I don't think `publicPath : auto` works.

I gives me `localhost:8000/auto` and my standalone remote app doesn't work (I can't even log in to the app).

i am working on vue3 so not using `webpack.config.js` but `vue-config.js`

I wanted to remove `localhost:8000/localhost:8000` two baseUrl from the remote app and also make sure I receive remoteEntry file in my host app.

**Reply by ScriptedAlchemy** [reply]
Auto should work unless some other plugin has a incompatibility with it. That auto feature is built right into webpack itself

**Comment by XiaofengXie16** [comment]
@probir1991 Would you mind to share a minimum reproducible? (either github repo or stackbliz )

**Reply by iamprobirdey** [reply]
I really wanted to. But it's private and official.
I will make some minimum stuff and get back.
Thanks for the comment. :)
